### PR TITLE
vars: add `pd_enable_follower_handle_region ` to support get region from pd follower

### LINKED ISSUE #49747: Allow PD follower handle region API
## Enhancement
ref https://github.com/tikv/pd/issues/7431

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: close #49747

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
image

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/sessionctx/variable/tidb_vars.go [hunk 1]
```diff
@@ -722,6 +722,9 @@ const (
 	// TiDBEnableTSOFollowerProxy indicates whether to enable the TSO Follower Proxy feature of PD client.
 	TiDBEnableTSOFollowerProxy = "tidb_enable_tso_follower_proxy"
 
+	// PDEnableFollowerRegionAPI indicates whether to enable the PD Follower handle region API.
+	PDEnableFollowerRegionAPI = "pd_enable_follower_region_api"
```
Comment (nolouch) [inline pkg/sessionctx/variable/tidb_vars.go:726]: How about `pd_enable_follower_handle_region`

File: pkg/domain/domain_sysvars.go [hunk 2]
```diff
@@ -69,6 +69,15 @@ func (do *Domain) setPDClientDynamicOption(name, sVal string) {
 			break
 		}
 		variable.EnableTSOFollowerProxy.Store(val)
+	case variable.PDEnableFollowerHandleRegion:
+		val := variable.TiDBOptOn(sVal)
+		// Note: EnableFollowerHandle is only used for region API now.
```
Comment (rleungx) [inline pkg/domain/domain_sysvars.go:74]: Should be EnableFollowerHandleRegion?

Comment (CabinfeverB) [reply pkg/domain/domain_sysvars.go:74]: As we discussed in [pr](https://github.com/tikv/pd/pull/7577), the variable names of pd client will not change for now. At present, we only need to define the names directly exposed to users in TiDB.

Comment (rleungx) [reply pkg/domain/domain_sysvars.go:74]: Got it, I remember that.

**Review by rleungx — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Pull request merged**